Add first pass of proper usage and documentation. Clean up errors (#3)
CapstoneThrone/amr-installer: README.md

# AMR installer

This repo is for preparing a new Raspberry PI 4 to become a AMR control module.

## What this doesn't do:
+ [Install the 64 bit version of Ubuntu 18.04.4 LTS on the SD card](https://ubuntu.com/download/raspberry-pi)
	+ *Note* If you've come to continue this project, it is very likely that the LTS version has changed. Just rolling with that should still work. Proceed with care!
	+ There's a separate guide for this. A link to it is in the "How to use" section
+ Get the PI on some functional network where it can download the updates

> 	+ Getting stuff on the stout network is silly. After some packages are installed we can use WPA2 Enterprise.... but before that we need some way to download packages.

Reviewer: This is picky, but capitalize Stout lol. Also, minor sentence change
```suggestion
	+ Getting stuff on the Stout network is silly. After some packages are installed we can connect to WPA2 Enterprise, but to install packages, we need an internet connection...
```

> 	+ Bringing the Pi to a non-stout lodging area would be advised. Otherwise a hostspot works well!

Reviewer: Again, going to be picky about Stout lol
```suggestion
	+ Bringing the Pi to a non-Stout lodging area would be advised. Otherwise a hotspot works well!
```

+ [apt](https://docs.ansible.com/ansible/latest/modules/apt_module.html)
+ [apt_key](https://docs.ansible.com/ansible/latest/modules/apt_key_module.html)
+ [command](https://docs.ansible.com/ansible/latest/modules/command_module.html#command-module)
+ [copy](https://docs.ansible.com/ansible/latest/modules/copy_module.html)
+ [file](https://docs.ansible.com/ansible/latest/modules/file_module.html)
+ [git](https://docs.ansible.com/ansible/latest/modules/git_module.html)
+ [loop](https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html)
+ [service](https://docs.ansible.com/ansible/latest/modules/service_module.html)
+ [template](https://docs.ansible.com/ansible/latest/modules/template_module.html)
+ [ufw](https://docs.ansible.com/ansible/latest/modules/ufw_module.html)
+ [user](https://docs.ansible.com/ansible/latest/modules/user_module.html)
+ [vault](https://docs.ansible.com/ansible/latest/user_guide/playbooks_vault.html)

It's recommended to quick read through files 1-5 to see what they do before running anything, just for an idea of what they will be doing to the system.

## How to use

###### Step 1

>  [Follow initial set-up to get pi running](https://liveuwstout.sharepoint.com/:w:/r/sites/AGVStoutSystem/CEE_Sensors_Design/_layouts/15/Doc.aspx?sourcedoc=%7B34122EEA-DB26-478D-9654-CB2B81CDFBEE%7D&file=Instructions%20for%20initial%20pi%20setup.docx&action=default&mobileredirect=true)

Reviewer: This link does not work. When I sign in with my Stout credentials, I am denied access. Consider placing documentation in more robust location (ie: this repository). Github storage is free :)

Author: I'll move it in here, and commit as @MaximUltimatum since he deserves the credit for that.

###### Step 2
 On the pi, we need to install ansible and check out this repo. Run the following commands.
```
$ sudo apt update
$ sudo apt install software-properties-common git
$ sudo apt-add-repository --yes --update ppa:ansible/ansible
$ sudo apt install ansible
$ git clone https://github.com/CapstoneThrone/amr-installer.git

> $ cd ./amr-installer

Reviewer: Is this supposed to work like this? Or did you mean something more like what I put in the suggestion?
```suggestion
$ cd /amr-installer
$ bash install.sh
```

Author: Yup, it works like that. It's checking out the repo in the current directory, making a folder named `amr-installer`

```
###### Step 3
We need to quick prepare and verify some settings. Please do the two following things:

> 1. Create a script in the `files` folder named `<robot name>-motd.sh` that outputs a fun pretty name. An example of how this file should look/work can be seen in `./files/example-motd.sh`. This can also optionally be an empty file, but ***it is required that this file exists***

Reviewer: How essential is this actually to the operation of the robot? I know you spent a lot of time on this @OneMoreByte, but this could add unnecessary complication to the setup unless your ansible plays depend on this file's existence.

Author: It does depend on it right now, but definitely agree this is a fluff detail that probably doesn't need to exist. I'll look into auto-creating the file if it isn't present so it can be optional, or removing that entirely.

> 2. Add stout credentials to `ros-secrets.yml`. To use StoutSecure, you need to use your stout credentials. What credentials go there is up to you. To add them, run

Reviewer: Are we going to assume the only use case for the robot is on a  Stout network? Is there a backup solution in the event someone is trying to get this robot running elsewhere (ie non-Stout network)? Is that even a use-case we need to consider?

Author: Right now, the ethernet port should be fine connecting to any network. As far as wireless, I guess I can see two solutions. We could make another playbook that would just make wireless connections, or a quick walkthrough on how to add a wireless connection with nmtui. I have no opinion on the importance. Maybe someone else feels strongly against? Otherwise I'll just run on one of those two.

Reviewer: If ethernet works anywhere, I'd be inclined to just go with that to save you some extra work. We'd have to mention that in the README. Anyone else have thoughts?

Reviewer 2: @ElegantDestruction If someone wants to use this somewhere that is not Stout I think that's outside the scope of this project. If they really want to do that they can modify the playbook.

```
# ansible-vault edit ./ros-secrets.yml
```
In this file you could optionally change the passwords for `ros` and `ros-adm`

###### Step 4
Each ansible playbook should be run in order. There are a total of 5 required ones, and represent logical groupings of tasks required to set the Pi up. When running each step, you will be asked for a `vault password`. This password is the same as what you were told to change the password to in step 1 and matches `ros-adm`'s password from the user guide.

> Run each of the following in order. Please replace `<robot name>` with the desired name of the robot. This process should take roughly an hour.

Reviewer: Any chance we could add a suggestion to chain these commands together with `&&`s so they don't have to sit there for an hour to watch it execute and run the next command? If you want that to ensure the scripts to succeed, ignore my comment, but I thought I'd mention it just in case.
```suggestion
Run each of the following in order. Please replace `<robot name>` with the desired name of the robot. This process should take roughly an hour. To save time, you can enter the commands all at once by typing them in a single line, inserting `&&` between each line, and pressing enter only after entering the last command.
```

Author: I think if we do that, it should just be a single script they run. I'm open to that. My only reservation is that it might make debugging harder. I'll work something up.

Reviewer 2: I like the way it is now. I think someone unfamiliar might be tempted to just run the script and walk away, and they'd miss an opportunity to see what went wrong rather than scrolling through logs for a few hours

Author: That was my concern too. I think I can make the script helpful in that sense though. I wanna give it a go :slightly_smiling_face:

> ```

Author: This probably needs more text. At the very least, expected output so people know when it went bad.
